src/toolbar.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+import { startHighlight } from './highlight';
+
+
+const toolbarStyle: React.HTMLAttributes<HTMLDivElement>["style"] = {
+  display: 'flex',
+  flexDirection: 'column',
+  position: 'fixed',
+  top: '100px',
+  right: '0px',
+  padding: '1px'
+};
+
+const toolbarContentStyle: React.HTMLAttributes<HTMLDivElement>["style"] = {
+  display: 'flex',
+  backgroundColor: '#d3d3d3',
+  flexDirection: 'column',
+  right: '0px',
+  border: '1px solid #757575'
+};
+
+const toolbarToggleStyle: React.HTMLAttributes<HTMLDivElement>["style"] = {
+  backgroundColor: '#d3d3d3',
+  width: '30px',
+  right: '0px',
+  padding: '5px'
+};
+
+const toolbarIconStyle: React.HTMLAttributes<HTMLDivElement>["style"] = {
+  opacity: '0.5',
+  width: '45px',
+  height: '45px'
+};
+
+class Toolbar extends Component {
```

Comment:
Please use function components over classes, and make sure to pass parameterized types as Component (and FC) are generics.

To deal with state, use the useState() react hook:
const [isActive, changeIsActive] = React.useState(false);
// isActive is a bool, changeIsActive is a (bool) => void that causes the component to re-render, with the new value of isActive

Reply:
const [isActive, toggleActive] = React.useReducer((e) => !e, false);